service/src/main/java/ai/chronon/service/handlers/FeaturesHandler.java

```diff
@@ -0,0 +1,135 @@
+package ai.chronon.service.handlers;
+
+import ai.chronon.online.JTry;
+import ai.chronon.online.JavaFetcher;
+import ai.chronon.online.JavaRequest;
+import ai.chronon.online.JavaResponse;
+import ai.chronon.service.model.GetFeaturesResponse;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.RequestBody;
+import io.vertx.ext.web.RoutingContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+import static ai.chronon.service.model.GetFeaturesResponse.Result.Status.Failure;
+import static ai.chronon.service.model.GetFeaturesResponse.Result.Status.Success;
+
+/**
+ * Concrete implementation of the GetFeatures endpoints. Supports loading groupBys and joins.
+ * Some notes on this:
+ * We currently support bulkGet lookups against a single groupBy / join. Attempts to lookup n different GroupBys / Joins
+ * need to be split up into n different requests.
+ * A given bulkGet request might result in some successful lookups and some failed ones. We return a 4xx or 5xx response
+ * if the overall request fails (e.g. we're not able to parse the input json, Future failure due to Api returning an error)
+ * Individual failure responses will be marked as 'Failed' however the overall response status code will be successful (200)
+ * The response list maintains the same order as the incoming request list.
+ * As an example:
+ * { results: [ {"status": "Success", "features": ...}, {"status": "Failure", "error": ...} ] }
+ */
+public class FeaturesHandler implements Handler<RoutingContext> {
+
+    public enum EntityType {
+        GroupBy,
+        Join
+    }
+
+    // PoJo to simplify transforming responses from the Fetcher to the final results form we return
+    private static class EntityKeyToValues {
+        public EntityKeyToValues(Map<String, Object> keys, JTry<Map<String, Object>> values) {
+            this.entityKeys = keys;
+            this.features = values;
+        }
+
+        public Map<String, Object> entityKeys;
+        public JTry<Map<String, Object>> features;
+    }
+
+    private static final Logger logger = LoggerFactory.getLogger(FeaturesHandler.class);
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    private final EntityType entityType;
+    private final JavaFetcher fetcher;
+
+    public FeaturesHandler(EntityType entityType, JavaFetcher fetcher) {
+        this.entityType = entityType;
+        this.fetcher = fetcher;
+    }
+
+    @Override
+    public void handle(RoutingContext ctx) {
+        String entityName = ctx.pathParam("name");
+        logger.info("Retrieving {} - {}", entityType.name(), entityName);
```

Comment:
should this be on debug level? 
```suggestion
        logger.debug("Retrieving {} - {}", entityType.name(), entityName);
```

Reply:
yeah lets drop down, else it will be noisy